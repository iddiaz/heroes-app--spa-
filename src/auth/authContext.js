import { createContext } from "react";

//createContext genera un componente
export const AuthContext = createContext();