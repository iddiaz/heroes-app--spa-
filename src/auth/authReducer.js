
import { types } from './../types/types';

// const state = {
//    name: 'Ivan',
//    logged: true
// }

// const loginAction = {

//    type: types.login,
//    payload: {
//       name: 'ivan',
//       email: 'correo@correo.com'
//    }

// }


export const authReducer =( state = {}, action )=>{
   switch ( action.type ) {
      case types.login:
            return {
               ...action.payload,
               logged: true
            }

      case types.logout:
            return {               
               logged: false
            }
     
   
      default:
         return state;
   }
}