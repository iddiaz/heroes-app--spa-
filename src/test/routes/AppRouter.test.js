import { AppRouter } from "../../routes/AppRouter";
import {mount} from 'enzyme';
import { AuthContext } from "../../auth/authContext";

describe('Pruebas en <AppRouter/>', ()=>{
   
   
   test('Debe mostrar el loggin si no está autenticado', ()=>{
      
      const contextValue = {
         user: {
            logged: false
         }
   
      }
      const wrapper = mount(
         <AuthContext.Provider value={contextValue} >
            <AppRouter/>
         </AuthContext.Provider>
      );

      // console.log ('=>>>>', wrapper.html() );

      expect (wrapper).toMatchSnapshot();
      expect(wrapper.find('h1').text().trim()).toBe('Login');

   }) 

   test('Debe mostrar el Componente de Marvel si está autenticado', ()=>{
      
      const contextValue = {
         user: {
            logged: true,
            name: 'Pepe'
         }   
      }

      const wrapper = mount(
         <AuthContext.Provider value={contextValue} >
            <AppRouter/>
         </AuthContext.Provider>
      );

      // console.log ('=>>>>', wrapper.html() );

      expect (wrapper).toMatchSnapshot();
      expect(wrapper.find('.navbar').exists()).toBeTruthy();

   })    

})