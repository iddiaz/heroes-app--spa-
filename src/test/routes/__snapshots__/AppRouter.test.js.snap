// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> Debe mostrar el Componente de Marvel si está autenticado 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
           
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        MarvelScreen
      </h1>
      <div
        className="row  row-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Spider Man"
                  className="card-img"
                  src="/assets/heroes/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Spider Man
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    Peter Parker
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Captain America"
                  className="card-img"
                  src="/assets/heroes/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Captain America
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    Steve Rogers
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Iron Man"
                  className="card-img"
                  src="/assets/heroes/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Iron Man
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    Tony Stark
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Thor"
                  className="card-img"
                  src="/assets/heroes/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Thor
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    Thor Odinson
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Hulk"
                  className="card-img"
                  src="/assets/heroes/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Hulk
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    Bruce Banner
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Wolverine"
                  className="card-img"
                  src="/assets/heroes/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Wolverine
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    James Howlett
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Daredevil"
                  className="card-img"
                  src="/assets/heroes/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Daredevil
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    Matthew Michael Murdock
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Daredevil #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Hawkeye"
                  className="card-img"
                  src="/assets/heroes/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Hawkeye
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    Clinton Francis Barton
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Cyclops"
                  className="card-img"
                  src="/assets/heroes/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Cyclops
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    Scott Summers
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    X-Men #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img"
                  src="/assets/heroes/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="cart-title"
                  >
                     
                    Silver Surfer
                     
                  </div>
                  <p
                    className="cart-text"
                  >
                     
                    Norrin Radd
                     
                  </p>
                </div>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  ver más
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <AppRouter/> Debe mostrar el loggin si no está autenticado 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
